receiver/awscontainerinsightreceiver/internal/host/ec2metadata.go

```diff
@@ -0,0 +1,96 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsec2metadata "github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	"go.uber.org/zap"
+)
+
+type Ec2metadata struct {
+	logger          *zap.Logger
+	refreshInterval time.Duration
+	instanceID      string
```

Comment:
I think we already have these as part of the resource detector

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go#L75

We don't need them here right?

Reply:
If it is used for detecting eks cluster name, then we might still need it, I think Jay removed it from resource detector because it only works for eks managed node group and cluster created using eksctl, i.e. not all k8s clusters on aws.

Though the ec2 meta data client logic can be moved to `internal/aws`

Reply:
The resource detector is a processor that doesn't provide a reliable way to get the eks cluster and there is no info about ebs volume. We had some logic in ebsvolume.go which require filesystem access to correctly get the device name: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8f261a14adcca1bec44c074d5ad2b639633f2fa2/receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go#L157-L167 

Considering the upcoming refactoring about the processor framework, I think it is a little risky to rely on resource detector. So I decided to put everything inside the receiver.